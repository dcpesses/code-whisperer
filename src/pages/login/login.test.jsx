/* eslint-env jest */
import {render} from '@testing-library/react';
import {vi} from 'vitest';
import Login from './index';
import React from 'react';

describe('Login', () => {
  describe('render', () => {
    test('Should render as expected', () => {
      vi.stubEnv('VITE_APP_TWITCH_CLIENT_ID', 'VITE_APP_TWITCH_CLIENT_ID');
      vi.stubEnv('VITE_APP_TWITCH_CLIENT_SECRET', 'VITE_APP_TWITCH_CLIENT_SECRET');
      vi.stubEnv('VITE_APP_REDIRECT_URI', 'VITE_APP_REDIRECT_URI');

      const {container} = render(
        <Login />
      );
      expect(container).toMatchSnapshot();

      vi.unstubAllEnvs();
    });
  });
});
