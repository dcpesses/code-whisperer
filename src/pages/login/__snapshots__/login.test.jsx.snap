// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Login > render > Should render as expected 1`] = `
<div>
  <div
    class="container full-pg fade-in"
    id="login-screen"
  >
    <div
      class="text-center pb-4"
    >
      <img
        alt="Logo"
        class="login-logo w-50 mx-auto mb-4"
        src="/code-whisperer/src/assets/new-logo.svg"
      />
    </div>
    <a
      class="btn btn-sm fs-2 py-2 px-3 rounded-4"
      href="https://id.twitch.tv/oauth2/authorize?client_id=VITE_APP_TWITCH_CLIENT_ID&response_type=code&scope=chat:read chat:edit moderation:read user:manage:whispers moderator:read:chatters user:read:moderated_channels channel:read:vips channel:read:editors moderator:manage:announcements user:read:subscriptions&redirect_uri=VITE_APP_REDIRECT_URI"
    >
      Log In With 
      <strong>
        Twitch
      </strong>
    </a>
    <div>
      <div
        class="text-center"
      >
        <small>
          v0.0.0
        </small>
      </div>
      <button
        class="btn btn-sm btn-link link-secondary text-decoration-none"
      >
        Having issues? 
        <span
          class="with-checkmark"
        >
          Reset application data
        </span>
      </button>
    </div>
    <p
      class="last-updated"
    >
      <small />
    </p>
  </div>
</div>
`;

exports[`Login > render > Should render with error during development 1`] = `
<div>
  <div
    class="container full-pg fade-in"
    id="login-screen"
  >
    <div
      class="text-center pb-4"
    >
      <img
        alt="Logo"
        class="login-logo w-50 mx-auto mb-4"
        src="/code-whisperer/src/assets/new-logo.svg"
      />
    </div>
    <div>
      <code
        class="small w-50 fs-6 mb-3"
      >
        Error: 
        Error Message
      </code>
    </div>
    <a
      class="btn btn-sm fs-2 py-2 px-3 rounded-4"
      href="https://id.twitch.tv/oauth2/authorize?client_id=VITE_APP_TWITCH_CLIENT_ID&response_type=code&scope=chat:read chat:edit moderation:read user:manage:whispers moderator:read:chatters user:read:moderated_channels channel:read:vips channel:read:editors moderator:manage:announcements user:read:subscriptions&redirect_uri=VITE_APP_REDIRECT_URI"
    >
      Log In With 
      <strong>
        Twitch
      </strong>
    </a>
    <div>
      <div
        class="text-center"
      >
        <small>
          v0.0.0
        </small>
      </div>
      <button
        class="btn btn-sm btn-link link-secondary text-decoration-none"
      >
        Having issues? 
        <span
          class="with-checkmark"
        >
          Reset application data
        </span>
      </button>
    </div>
    <p
      class="last-updated"
    >
      <small />
    </p>
  </div>
</div>
`;

exports[`Login > render > Should render without error in production 1`] = `
<div>
  <div
    class="container full-pg fade-in"
    id="login-screen"
  >
    <div
      class="text-center pb-4"
    >
      <img
        alt="Logo"
        class="login-logo w-50 mx-auto mb-4"
        src="/code-whisperer/src/assets/new-logo.svg"
      />
    </div>
    <div>
      <code
        class="small w-50 fs-6 mb-3"
      >
        Error: 
        Error Message
      </code>
    </div>
    <a
      class="btn btn-sm fs-2 py-2 px-3 rounded-4"
      href="https://id.twitch.tv/oauth2/authorize?client_id=VITE_APP_TWITCH_CLIENT_ID&response_type=code&scope=chat:read chat:edit moderation:read user:manage:whispers moderator:read:chatters user:read:moderated_channels channel:read:vips channel:read:editors moderator:manage:announcements user:read:subscriptions&redirect_uri=VITE_APP_REDIRECT_URI"
    >
      Log In With 
      <strong>
        Twitch
      </strong>
    </a>
    <div>
      <div
        class="text-center"
      >
        <small>
          v0.0.0
        </small>
      </div>
      <button
        class="btn btn-sm btn-link link-secondary text-decoration-none"
      >
        Having issues? 
        <span
          class="with-checkmark"
        >
          Reset application data
        </span>
      </button>
    </div>
    <p
      class="last-updated"
    >
      <small />
    </p>
  </div>
</div>
`;
